package cpuvisor;

import "visor_config.proto";

extend visor.Config {
  optional CaffeConfig caffe_config = 100;
}

extend visor.PreprocConfig {
  optional string neg_im_base_path = 101;
  optional string neg_im_paths = 102;
  optional string neg_index_file = 105;

  optional DataAugType data_aug_type_preproc = 120;
}

extend visor.ServiceConfig {
  optional DataAugType data_aug_type_srv = 120;
}

extend visor.ServerConfig {
  optional string image_cache_path = 110;
  optional string rlist_cache_path = 111;
}

enum DataAugType {
  DAT_NONE = 0;
  DAT_ASPECT_CORNERS = 1;
}

enum CaffeMode {
  CM_CPU = 0;
  CM_GPU = 1;
}

message CaffeConfig {
  optional string param_file = 1;
  optional string model_file = 2;
  optional string mean_image_file = 3;
  optional string output_blob_name = 4 [default = "fc7"];
  optional DataAugType data_aug_type = 5 [default = DAT_NONE];
  optional CaffeMode mode = 6 [default = CM_CPU];

  optional bool use_rgb_images = 16 [default = false];
  optional uint32 netpool_sz = 17 [default = 1];
}
